legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalIngestorAbstract.java

```diff
@@ -470,6 +472,24 @@ public void dedupAndVersion()
                 }
             }
 
+            // Error Check for PK Duplicates: if versioning = No Versioning (fail on pk dups), Fail the job if count > 1
+            if (dedupAndVersionErrorSqlTypeSqlPlanMap.containsKey(MAX_PK_DUPLICATES))
+            {
+                List<TabularData> result = executor.executePhysicalPlanAndGetResults(dedupAndVersionErrorSqlTypeSqlPlanMap.get(MAX_PK_DUPLICATES));
+                Optional<Object> obj = getFirstColumnValue(getFirstRowForFirstResult(result));
+                Optional<Long> maxPkDuplicatesValue = retrieveValueAsLong(obj.orElse(null));
+                if (maxPkDuplicatesValue.isPresent() && maxPkDuplicatesValue.get() > 1)
+                {
+                    // Find the pk-duplicate rows
+                    TabularData duplicatePkRows = executor.executePhysicalPlanAndGetResults(dedupAndVersionErrorSqlTypeSqlPlanMap.get(PK_DUPLICATE_ROWS)).get(0);
+                    String errorMessage = "Encountered duplicate primary keys, Failing the batch as Fail on Duplicate Primary Keys is selected";
+                    LOGGER.error(errorMessage);
```

Comment:
Maybe the logger also contain some details about the running pipeline? (which dataset/datastore/ etc) will be useful when debugging

Reply:
It seems like the library loggers traditionally do not log these info. LH already logs it so we already have this information for debugging